test/suites/dev-tanssi-relay/opengov/test-whitelist-track-referenda.ts

```diff
@@ -0,0 +1,247 @@
+                    }
+                );
+
+                // 2. Create whitelisted call referendum proposal for the whitelisted track
+                const whitelistedCallReferendumProposalTx = api.tx.referenda.submit(
+                    {
+                        Origins: "WhitelistedCaller",
+                    },
+                    {
+                        Lookup: {
+                            Hash: whitelistedCallDispatchTx.method.hash.toHex(),
+                            len: whitelistedCallDispatchTx.method.encodedLength,
+                        },
+                    },
+                    { After: "1" }
+                );
+                const whitelistedCallReferendumProposalBlock = await context.createBlock(
+                    await whitelistedCallReferendumProposalTx.signAsync(charlie)
+                );
+
+                // 3. Extract referendum index
+                expect(whitelistedCallReferendumProposalBlock.result?.successful).to.be.true;
+
+                const proposalIndex = (
+                    whitelistedCallReferendumProposalBlock.result?.events
+                        .find((e) => e.event.method === "Submitted")
+                        .event.toHuman().data as unknown as SubmittedEventDataType
+                ).index;
+                expect(proposalIndex).to.not.be.undefined;
+
+                // 4. Place decision deposit for the proposal
+                const depositSubmitBlock = await context.createBlock(
+                    await api.tx.referenda.placeDecisionDeposit(proposalIndex).signAsync(charlie)
+                );
+                expect(depositSubmitBlock.result?.successful).to.be.true;
+
+                // 5. Vote
+                for (const voter of [alice, bob, dave]) {
+                    const voteSubmit = await context.createBlock(
+                        await api.tx.convictionVoting
+                            .vote(proposalIndex, {
+                                Standard: { vote: { aye: true, conviction: "None" }, balance: 999999000000000000n },
+                            })
+                            .signAsync(voter)
+                    );
+                    expect(voteSubmit.result?.successful).to.be.true;
+                }
+
+                // 6. Wait for the referendum to be decided
+                const missingReferendumDecisionEvents = await waitForReferendumDecision(context, api);
+
+                // Check if all the events happened in the specified order.
+                expect(missingReferendumDecisionEvents).toEqual([]);
+
+                // 7. Confirm proxy is not added
+                const proxyInfo = await api.query.proxy.proxies(dave.address);
+                const [delegates] = proxyInfo.toJSON() as { delegate: string }[][];
+                const added = delegates.find((d: any) => d.delegate === delegate);
+                expect(added).not.to.exist;
+            },
+        });
+    },
```

Comment:
Do we also want to add a smoke test to see if any proposals were submitted to the whitelisted track?

Reply:
yes, we might need that. but that would be specially for the root track. and the root track will very likely have a sufficiently good acceptance criteria